### PR TITLE
Update CI Process

### PR DESCRIPTION
**Pull request by BrendanDavies**
What:
Remove need to check-in `dist` folder
Add Readme/Changelog
Upgrade node (and npm)

Why:
To make contributing easier

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Changelog
```
Comment (fghilardi) [inline CHANGELOG.md]: 👏 thanks for this

File: README.md [hunk 2]
```diff
@@ -2,9 +2,63 @@
...
+
+Settings of the network request that will be passed along to any network request.
+
+Additionally `hypergard` implements a timeout around any network fetch, which will default to `60000` milliseconds
+
+```js
+{
+  // Headers to be added to each network fetch
+  headers: {
+    ['X-Custuom-Header']: "value",
```
Comment (fghilardi) [inline README.md]: `X-Custom-Header`  ?

**Review by fghilardi — COMMENTED** [review]
Do we not need the `.npmrc` any longer?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hypergard",
-  "version": "5.2.0",
+  "version": "6.0.0",
```
Comment (fghilardi) [inline package.json]: would you be opposed to adding an an `engines` section here to indicate which node + npm version to run?

```json
  "engines": {
    "node": ">=10.0.0",
    "npm": ">=6.0.0"
  },```

probably does it

**Comment by BrendanDavies** [comment]
Re: the npmrc, once I upgraded node, I was able to publish using the key in the travis file alone.

See https://www.npmjs.com/package/hypergard/v/6.0.0-rc1 That was publish from the CI branch tag.

**Review by fghilardi — APPROVED** [review]

**Pull request merged**